contacts.py

```diff
@@ -400,6 +401,12 @@ def admin_status(self, data, args):
         }
         return ret
 
+    # Return a matching date - see issue#57 for discussion of a valid date
+    # Essentially is date < now because of confusion with items arriving the same second,
+    # And is since <= date so that passing back now will get any events that happened on that second
+    def _good_date(self, date, since = None, now = None):
+        return (not since) or (since <= date) and ((not now) or (date < calendar.timegm(now)))
```

Comment:
can you please fix parens so ordering is clear (ie is `A or B and C` `(A or B) and C` or `A or (B and C)`)